prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -255,16 +257,22 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
      * Get a credential using PrivateKey to generate signed JWT authentication token (senderToken)
      * @param receiver the fullName of the receiver is the label of the credential
      */
-    fun lookupTwoLeggedCredential(receiver: Receiver): RestCredential {
+    fun lookupTwoLeggedCredential(receiver: Receiver, jwtParams: Map<String, String> = emptyMap()): RestCredential {
         val credential = lookupDefaultCredential(receiver)
         val privateKey = AuthUtils.readPrivateKeyPem((credential as UserApiKeyCredential).apiKey)
-        val senderToken = AuthUtils.generateOrganizationToken(
-            Organization(
-                receiver.name, receiver.fullName,
-                Organization.Jurisdiction.FEDERAL, null, null, null, null, null
-            ),
-            "", privateKey, ""
-        )
+        val senderToken = if (jwtParams.isEmpty()) {
+            AuthUtils.generateOrganizationToken(
+                Organization(
+                    receiver.name, receiver.fullName,
+                    Organization.Jurisdiction.FEDERAL, null, null, null, null, null
+                ),
+                "", privateKey, ""
+            )
+        } else {
+            val issuer = jwtParams["iss"] ?: receiver.name
+            val audience = jwtParams["aud"] ?: ""
```

Comment:
I'm trying to figure out if the issuer and the audience is sensitive information and should be securely stored along the receiver credentials of having it on the receiver settings is fine?
Would it be better to have a new auth-type called "two-legged-jwt" and use that instead of checking if jwt-params is not empty?
Is it safe to default to the receiver.name  if issuer is empty?

Reply:
These are excellent questions @victor-chaparro . Thank you for your patience. We are currently determining if the issuer field constitutes sensitive information. If it is sensitive, it should not be included in the repository. For now, the issuer is used only for testing EPIC's Sandbox environment and will differ in the production environment. If we retain this configuration, we can update the settings file via the UI or CLI and include the `issuer` there.

Regarding the `audience`, EPIC has it disclosed on their public [documentation ](https://fhir.epic.com/Documentation?docId=oauth2&section=BackendOAuth2Guide), so it is not considered sensitive information. 

We default the issuer to `receiver.name` to maintain pre-existing functionality and behavior. `AuthUtils.generateOrganizationToken` uses  `receiver.name`  for the `issuer` and leaves the `audience` blank. If the jwtParams map is empty, we revert to the previous settings as part of the null handling process.

I am open to the suggestion of creating a new `authType`. However, adding a map for JWT customization keeps this feature within the `two-legged` `authType` without creating a duplicate `authType` with an additional feature. This approach also allows for future extension to other `authType` values that generate JWTs. For now, it will be part of the `two-legged` `authType`, but we can extend it to other types if needed. Thoughts?

Reply:
Thanks for the details responses @jorg3lopez. I'll go ahead and approve this.